ios/objective-c/AmpliObjectiveCSampleApp/AmpliObjectiveCSampleApp/Ampli/Ampli.h

```diff
@@ -0,0 +1,607 @@
+ * To update run 'ampli pull obj-c-ampli'
+ *
+ * Required dependencies: Amplitude-iOS^8.6.0
+ * Tracking Plan Version: 0
+ * Build: 1.0.0
+ *
+ * [View Tracking Plan](https://data.amplitude.com/test-codegen/Test%20Codegen/events/main/latest)
+ *
+ * [Full Setup Instructions](https://data.amplitude.com/test-codegen/Test%20Codegen/implementation/obj-c-ampli)
+ */
+#import <Foundation/Foundation.h>
+#import "Amplitude.h"
+
+typedef NSMutableDictionary<NSString *, id> MiddlewareExtra;
+
+typedef NS_ENUM(NSInteger, AmpliEnvironment) {
+  development,
+  production
+};
+
+@interface Event: NSObject
+@property (nonatomic, strong, readonly) NSString * _Nonnull eventType;
+@property (nonatomic, strong, readonly) NSDictionary * _Nullable eventProperties;
+@end
+
+@interface LoadClientOptions: NSObject
+@property (nonatomic, strong, readwrite) NSString * _Nullable apiKey;
+@property (nonatomic, strong, readwrite) Amplitude * _Nullable instance;
++ (instancetype _Nonnull)initWithApiKey: (NSString *_Nonnull) apiKey;
++ (instancetype _Nonnull)initWithAmplitudeInstance: (Amplitude *_Nonnull)instance;
+@end
+
+@interface LoadOptions: NSObject
+@property (nonatomic, readwrite) AmpliEnvironment environment;
+@property (nonatomic, readwrite) BOOL disabled;
+@property (nonatomic, readwrite) LoadClientOptions * _Nullable client;
++ (instancetype _Nonnull)initWithApiKey:(NSString *_Nonnull) apiKey;
++ (instancetype _Nonnull)initWithAmplitudeInstance:(Amplitude *_Nonnull)instance;
++ (instancetype _Nonnull)initWithEnvironment:(AmpliEnvironment)environment;
++ (instancetype _Nonnull)initWithDisabled:(BOOL) disabled;
++ (instancetype _Nonnull)initWithEnvironmentAndDisabled:(AmpliEnvironment)environment disabled:(BOOL) disabled;
++ (instancetype _Nonnull)initWithClient:(LoadClientOptions *_Nonnull)client;
+@end
+
+@interface EventOptions : NSObject
+@property (nonatomic, strong) NSString * _Nullable deviceId;
+@property (nonatomic, strong) NSString * _Nullable userId;
++ (instancetype _Nonnull)initWithDeviceId:(NSString *_Nonnull)deviceId;
++ (instancetype _Nonnull)initWithDeviceIdAndUserId:(NSString *_Nullable)deviceId userId:(NSString *_Nullable)userId;
+@end
+
+@class EventProperties;
+@class ContextProperties;
+@class EventMaxIntForTestProperties;
+@class EventNoPropertiesProperties;
+@class EventObjectTypesProperties;
+@class EventWithAllPropertiesProperties;
+@class RequiredEnum;
+@class EventWithArrayTypesProperties;
+@class EventWithConstTypesProperties;
+@class EventWithDifferentCasingTypesProperties;
+@class EnumCamelCase;
+@class EnumPascalCase;
+@class EnumSnakeCase;
+@class EnumWithSpace;
+@class EventWithEnumTypesProperties;
+@class OptionalEnum;
+@class RequiredEnumEnum;
+@class EventWithOptionalArrayTypesProperties;
+@class EventWithOptionalPropertiesProperties;
+@class GroupProperties;
+@class IdentifyProperties;
+
+NS_ASSUME_NONNULL_BEGIN
+
+#pragma mark - Boxed enums
+
+/// Event 2 Property - Enum
+@interface RequiredEnum : NSObject
```

Comment:
Andrey is also finding issues with Enums in Kotlin (along with some naming concerns) so I think we should create a fork of Quicktype like we talked about so we can implement these how we want. Sounds good to you?